src/app/modules/group/components/user-list/user-list.component.ts

```diff
@@ -8,43 +10,48 @@ import { GetGroupMembersService, Member } from '../../http-services/get-group-me
   templateUrl: './user-list.component.html',
   styleUrls: [ './user-list.component.scss' ]
 })
-export class UserListComponent implements OnChanges, OnDestroy {
+export class UserListComponent implements OnInit {
 
   @Input() group? : Group;
   state: 'loading' | 'error' | 'empty' | 'ready' = 'loading';
+  currentSort: string[] = [];
 
   members: Member[] = [];
 
-  private subscription?: Subscription;
+  private dataFetching = new BehaviorSubject<'Reload'|'Sort'>('Reload');
 
   constructor(private getGroupMembersService: GetGroupMembersService) { }
 
-  ngOnChanges(_changes: SimpleChanges): void {
-    this.reloadData();
+  ngOnInit(): void {
+    this.dataFetching.pipe(
+      switchMap(mode => {
+        if (this.group) {
+          if (mode === 'Reload')
+            this.state = 'loading';
+
+          return this.getGroupMembersService.getGroupMembers(this.group.id, this.currentSort);
+        } else {
+          throw new Error('group is null');
+        }
+      })
+    ).subscribe(
+      members => {
+        this.members = members;
+
+        if (this.members.length === 0) this.state = 'empty';
+        else this.state = 'ready';
+      },
+      _err => {
+        this.state = 'error';
+      });
   }
 
-  private reloadData(): void {
-    if (this.group) {
-      this.state = 'loading';
-      this.subscription?.unsubscribe();
-      this.subscription = this.getGroupMembersService.getGroupMembers(this.group.id)
-        .subscribe(
-          members => {
-            this.members = members;
-
-            if (this.members.length === 0) this.state = 'empty';
-            else this.state = 'ready';
-          },
-          _err => {
-            this.state = 'error';
-          }
-        );
-    } else {
-      this.state = 'error';
-    }
-  }
+  onCustomSort(event: SortEvent): void {
+    const sortMeta = event.multiSortMeta?.map(meta => (meta.order === -1 ? `-${meta.field}` : meta.field));
 
-  ngOnDestroy(): void {
-    this.subscription?.unsubscribe();
+    if (sortMeta && JSON.stringify(sortMeta) !== JSON.stringify(this.currentSort)) {
```

Comment:
(Having less state is always better (less inconsistency, ...))
I don't think storing `currentSort` is useful in anyway, it just has to be passed to the dataFetching, but not to be stored.
While quickly checking the impact, we can see it means that if `group` change, the sort is not kept, but that seems to be a better idea anyway. Note there is a "reset" method to be used on p-table especially for this case.

Reply:
The function `onCustomSort` is called when the data is refreshed (I don't know why) but if I get rid of the `currentSort` variable, the data will be reloaded continuously, so I kept it to not enter the if statement if the data is already sorted.

Reply:
From what I see from ngprime code, when there is a change in the content, the reapply the sort... which makes sense if it is a static sort. I do not see any other easy solution...